### PR TITLE
[Improve] [Seatunnel-Engine] remove `seatunnel-api` from engine storage.

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by liugddx** [comment]
#3792

**Comment by Hisoka-X** [comment]
Hi, didn't find the code which call SPI method after code change.

File: seatunnel-engine/seatunnel-engine-common/src/main/java/org/apache/seatunnel/engine/common/utils/FactoryUtil.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class FactoryUtil<T> {
+
+    public static <T> T discoverFactory(ClassLoader classLoader, Class<T> factoryClass, String factoryIdentifier) {
+        try {
+            final List<T> result = new LinkedList<>();
+            ServiceLoader.load(factoryClass, classLoader)
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-common/src/main/java/org/apache/seatunnel/engine/common/utils/FactoryUtil.java:40]: @Hisoka-X  it's here. I just simplified a FactoryUtil over.

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-common/src/main/java/org/apache/seatunnel/engine/common/utils/FactoryUtil.java:40]: I mean the code call `discoverFactory`

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-common/src/main/java/org/apache/seatunnel/engine/common/utils/FactoryUtil.java:40]: > I mean the code call `discoverFactory`

This is the case, before calling `discoverFactory`, the method name has not been modified.

**Review by Hisoka-X — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**